api/v1alpha1/openapi.yaml

```
@@ -2569,6 +2569,42 @@ components:
               - filePath
           required:
           - httpRef
+    ApplicationSpec:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/ApplicationEnvVars'
+        - type: object
+          properties:
+            name:
+              type: string
+              description: "The name of the application"
+        - oneOf:
+            - $ref: '#/components/schemas/ImageApplicationProvider'
+            # extend application providers
```

```suggestion
            # extend application providers
```

ok that makes sense, I hope we will be able to add other application providers (like coming from a directory in the device that holds podman-compose, helm charts or other stuff i.e)

---

Right this is MVP for applications and with this we can demo the flow e2e. From there we will branch out.